dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/DynatraceMetricExporter.java

```diff
@@ -102,32 +130,123 @@ public CompletableResultCode export(@Nonnull Collection<MetricData> metrics) {
+        case HISTOGRAM:
+          metricLines.addAll(serializer.createDoubleHistogramLines(metric));
+          break;
+        default:
+          logger.warning(
+              String.format(
+                  "Tried to serialize metric of type %s. The Dynatrace metrics exporter does not handle metrics of that type at the time.",
+                  metric.getType().toString()));
+          break;
+      }
+    }
+
+    return metricLines;
+  }
+
   @VisibleForTesting
   protected CompletableResultCode export(
       Collection<MetricData> metrics, HttpURLConnection connection) {
-    String mintMetricsMessage = MetricAdapter.toMint(metrics).serialize();
-    logger.log(Level.FINEST, "Exporting: {0}", mintMetricsMessage);
-    try {
-      connection.setRequestMethod("POST");
-      connection.setRequestProperty("Accept", "*/*; q=0");
-      connection.setRequestProperty("Authorization", "Api-Token " + apiToken);
-      connection.setRequestProperty("Content-Type", "text/plain; charset=utf-8");
-      connection.setDoOutput(true);
-      try (final OutputStream outputStream = connection.getOutputStream()) {
-        outputStream.write(mintMetricsMessage.getBytes(StandardCharsets.UTF_8));
+
+    List<String> metricLines = serializeToMetricLines(metrics);
+    for (List<String> partition :
+        Lists.partition(metricLines, DynatraceMetricApiConstants.getPayloadLinesLimit())) {
+      CompletableResultCode resultCode;
+      String joinedMetricLines = Joiner.on('\n').join(partition);
+
+      if (logger.isLoggable(Level.FINER)) {
+        logger.finer(String.format("Exporting metrics:\n%s", joinedMetricLines));
       }
-      int code = connection.getResponseCode();
-      if (code != 202) {
-        logger.log(Level.WARNING, "Received error code {0} from server", code);
-        return CompletableResultCode.ofFailure();
+      try {
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Accept", "*/*; q=0");
+        if (this.apiToken != null) {
+          connection.setRequestProperty("Authorization", "Api-Token " + apiToken);
+        }
+        connection.setRequestProperty("Content-Type", "text/plain; charset=utf-8");
+        connection.setRequestProperty("User-Agent", "opentelemetry-metric-java");
+        connection.setDoOutput(true);
+        try (final OutputStream outputStream = connection.getOutputStream()) {
+          outputStream.write(joinedMetricLines.getBytes(StandardCharsets.UTF_8));
+        }
+        int code = connection.getResponseCode();
+        if (code < 400) {
+          String response =
+              CharStreams.toString(
+                  new InputStreamReader(connection.getInputStream(), Charsets.UTF_8));
+          resultCode = handleSuccess(code, metricLines.size(), response);
+        } else {
+          String response =
+              CharStreams.toString(
+                  new InputStreamReader(connection.getErrorStream(), Charsets.UTF_8));
+          logger.log(
```

Comment:
```suggestion
        int code = connection.getResponseCode();
        String response =
            CharStreams.toString(
                new InputStreamReader(connection.getInputStream(), Charsets.UTF_8));
        if (code < 400) {
          resultCode = handleSuccess(code, metricLines.size(), response);
        } else {
          logger.log(
```

Reply:
Unfortunately, this does not work. Non-error codes use `getInputStream()` while error codes use `getErrorStream()`